code/modules/research/data-analysis/decryption.dm

```diff
@@ -0,0 +1,171 @@
+/obj/machinery/computer/decryption
+	name="Data-Decryption Console"
+	desc="This console is used to analyse and decrypt data found from various sources"
+	icon = 'icons/obj/computer.dmi'
+	icon_screen = "dna"
+	icon_keyboard = "med_key"
+	density = 1
+	var/obj/item/weapon/book/databook/databook=null
+	// Currently loaded databook.
+	var/init = 1
+
+	var/A
```

Comment:
These are used to store the currently selected puzzle parts selected by the player, while they wait to be tested against the keyfragments. Altough looking back on this, it sure is messy as hell. I implemented that system so I could have the template wait with updating the circle colours untill the solve button was pushed. When I tried this with just key fragments and unsolved fragments, it would instantly change colour on selecting a correct answer.